python/lsst/ts/mtdome/mock_llc/mock_motion/shutter_motion.py

```diff
@@ -0,0 +1,343 @@
+# This file is part of ts_mtdome.
+#
+# Developed for the Vera Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = [
+    "ShutterMotion",
+    "CLOSED_POSITION",
+    "NUM_MOTORS_PER_SHUTTER",
+    "OPEN_POSITION",
+    "SHUTTER_SPEED",
+]
+
+import logging
+import math
+from typing import Tuple
+
+from ...enums import LlcMotionState
+from .base_llc_motion import BaseLlcMotion
+
+# The number of motors per shutter.
+NUM_MOTORS_PER_SHUTTER = 2
+
+# The shutter is 0% open.
+CLOSED_POSITION = 0.0
+# The shutter is 100% open.
+OPEN_POSITION = 100.0
+# The shutter speed (%/s). This is an assumed value such that the shutter opens
+# or closes in 10 seconds.
+SHUTTER_SPEED = 10.0
+
+
+class ShutterMotion(BaseLlcMotion):
```

Comment:
Surely this is not the only LlcMotion system that has no crawl velocity. Please try to factor out most of this code into a new base class for such subsystems.

Another option is to expand the base class to handle systems that cannot crawl. In such systems it is an error to request a non-zero crawl velocity and the motion state can never be "crawling".

Reply:
I created DM-35913 for this.